client/component/src/ConnectedPlot.tsx

```diff
@@ -394,12 +393,12 @@ function ConnectedPlot(props: ConnectedPlotProps) {
     });
   };
 
-  const updateSelection = (
+  const cUpdateSelection = (
     selection: SelectionBase | null,
     broadcast = true,
     clear = false
   ) => {
-    const id = addSelection(selection, clear);
+    const id = updateSelection(selection, clear);
```

Comment:
This line is causing the issue with selections not being cleared correctly.
updateSelection takes three parameters: `selection, broadcast = false, clear = false` so the `updateSelection(selection, clear)` call leaves `clear` set to `false`.